### PR TITLE
Fix comparison kernel benchmarks

### PR DESCRIPTION
**Pull request by samuelcolvin**
(Sorry, I'll stop creating PRs now I promise 🙏)

# Rationale for this change
 
See https://github.com/apache/arrow-rs/pull/6128#issuecomment-2254513573 for an explanation of the rationale - TL;DR I think varied lengths strings with a max length of 400 is more representative than all strings having 4 characters.

Happy to change the 400 to 100 or some other number if you'd prefer.

# What changes are included in this PR?

* change the benchmark utility function `create_string_array` to return an array of strings of varied length (`0..400`), not a fixed length of 4
* fix`comparison_kernel` bencmarks so the like expressions use the correct literals for "starts with" and "ends with"

# Are there any user-facing changes?

no.

### CODE REVIEW

**Comment by alamb** [comment]
> (Sorry, I'll stop creating PRs now I promise 🙏)

No please keep them coming. I am sorry I don't have the time to devote to reviewing them as fast as I would like to

I am also sort of hoping some of the other maintainers will step up too

File: arrow/benches/comparison_kernels.rs [hunk 1]
```diff
@@ -215,11 +215,11 @@ fn add_benchmark(c: &mut Criterion) {
     });
 
     c.bench_function("like_utf8 scalar ends with", |b| {
-        b.iter(|| bench_like_utf8_scalar(&arr_string, "xxxx%"))
+        b.iter(|| bench_like_utf8_scalar(&arr_string, "%xxxx"))
```
Comment (alamb) [inline arrow/benches/comparison_kernels.rs:218]: 💯  for fixing the inversion

File: arrow/src/util/bench_util.rs [hunk 2]
```diff
@@ -108,12 +108,33 @@ where
...
 }
 
-/// Creates an random (but fixed-seeded) array of a given size and null density
+/// Creates a random (but fixed-seeded) array of a given size and null density
 pub fn create_string_array<Offset: OffsetSizeTrait>(
     size: usize,
     null_density: f32,
 ) -> GenericStringArray<Offset> {
-    create_string_array_with_len(size, null_density, 4)
+    create_string_array_with_max_len(size, null_density, 400)
```
Comment (alamb) [inline arrow/src/util/bench_util.rs:116]: Maybe worth a small comment about using relatively long strings

**Review by alamb — APPROVED** [review]
Makes sense to me -- thank you @samuelcolvin

File: arrow/src/util/bench_util.rs (hunk 2 above)
Comment (samuelcolvin) [reply arrow/src/util/bench_util.rs:118]: done.

**Comment by alamb** [comment]
Thanks again @samuelcolvin

**Pull request merged**